multiqc/modules/hicpro/hicpro.py

```diff
@@ -0,0 +1,392 @@
+             'title': '% Valid Pairs Unique',
+             'description': 'Percent of valid pairs over reported ones after duplicates removal (%)',
+             'max': 100,
+             'min': 0,
+             'suffix': '%',
+             'scale': 'YlGn'
+             #'modify': lambda x: 100 - x
+         }
+     
+         self.general_stats_addcols(self.hicpro_data, headers, 'HiC-Pro')
+
+
+    def hicpro_mapping_chart (self):
+        """ Generate the HiC-Pro Aligned reads plot """
+
+        # Specify the order of the different possible categories
+        keys = OrderedDict()
+        keys['Full_Alignments_Read']   = { 'color': '#005ce6', 'name': 'Full reads Alignments' }
+        keys['Trimmed_Alignments_Read'] = { 'color': '#3385ff', 'name': 'Trimmed reads Alignments' }
+        keys['Failed_To_Align_Read']     = { 'color': '#a9a2a2', 'name': 'Failed To Align' }
+    
+        # Construct a data structure for the plot - duplicate the samples for read 1 and read 2
+        data = {}
+        for s_name in self.hicpro_data:
+            data['{} Read 1'.format(s_name)] = {}
+            data['{} Read 2'.format(s_name)] = {}
+            data['{} Read 1'.format(s_name)]['Full_Alignments_Read'] = self.hicpro_data[s_name]['global_R1']
+            data['{} Read 2'.format(s_name)]['Full_Alignments_Read'] = self.hicpro_data[s_name]['global_R2']
+            data['{} Read 1'.format(s_name)]['Trimmed_Alignments_Read'] = self.hicpro_data[s_name]['local_R1']
+            data['{} Read 2'.format(s_name)]['Trimmed_Alignments_Read'] = self.hicpro_data[s_name]['local_R2']
+            data['{} Read 1'.format(s_name)]['Failed_To_Align_Read'] = int(self.hicpro_data[s_name]['total_R1']) - int(self.hicpro_data[s_name]['mapped_R1'])
+            data['{} Read 2'.format(s_name)]['Failed_To_Align_Read'] = int(self.hicpro_data[s_name]['total_R2']) - int(self.hicpro_data[s_name]['mapped_R2'])
+
+        # Config for the plot
+        config = {
+            'id': 'hicpro_mapping_stats_plot',
+            'title': 'HiC-Pro: Mapping Statistics',
+            'ylab': '# Reads',
+            'cpswitch_counts_label': 'Number of Reads'
+        }
+
+        return bargraph.plot(data, keys, config)
+
+    def hicpro_pairing_chart (self):
+        """ Generate Pairing chart """
+
+        # Specify the order of the different possible categories
+        keys = OrderedDict()
+        keys['Unique_Pairs'] = { 'color': '#005ce6', 'name': 'Uniquely Aligned' }
+        keys['Low_Qual_Pairs'] = { 'color': '#ff8533', 'name': 'Low Quality' }
+        keys['Singleton_Pairs'] = { 'color': ' #ff9933', 'name': 'Singleton' }
+        keys['Multi_Pairs'] = { 'color': '#e67300', 'name': 'Multi Aligned' }
+        keys['Failed_To_Align_Pairs'] = { 'color': '#a9a2a2', 'name': 'Failed To Align' }
+    
+        # Construct a data structure for the plot - duplicate the samples for read 1 and read 2
+        data = {}
+        for s_name in self.hicpro_data:
```

Comment:
I'm not really sure why you need this block. I think you can just submit `self.hicpro_data` directly to the plotting function and get essentially the same result (the `keys` dict above specifies which data to plot, the rest is ignored).

Reply:
...exactly as you've done in the `hicpro_filtering_chart` plotting function  👍

Reply:
No - you're updating the dictionary keys here which isn't visible. You're setting the plot labels just above (line [294](https://github.com/ewels/MultiQC/pull/785/files#diff-f9fdf5a489f4b1f5aefdd0e79b324f18R294)).

Reply:
fixed